src/utils/Api.js

```diff
@@ -0,0 +1,67 @@
+class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) return res.json();
+    return Promise.reject(`Error: ${res.status} ${res.statusText}`);
+  }
+
+  getAppInfo() {
+    return Promise.all([this.getInitialCards(), this.getUserInfo()]);
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    }).then(this._checkResponse);
```

Comment:
**COULD BE IMPROVED**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.

Reply:
Hello. Thank you for the review Mr. Gennady. 

Please clarify the following.

"You need to insert the profile data into the inputs when you open the profile modal to let the user edit the data and submit it."

Please explain precisely what you mean by insert. Are we generally speaking about the users ability to use the app? The edit profile button and modal seems to work on my end.